General Docs Improvements (#1194)
CloudFire-LLC/cloudfire: docs/docs/administer/troubleshoot.mdx

---
title: Troubleshoot
sidebar_position: 6

> description: Troubleshoot common connectivity and configuration issues with Firezone's WireGuard®-based remote access platform.

Reviewer: Consistency -- `secure access` in some places but `remote access` in others.

```suggestion
description: Troubleshoot common connectivity and configuration issues with Firezone's WireGuard®-based secure access platform.
```

---

This guide documents common configuration and connectivity issues. For
any problems that arise, a good first bet is to check the Firezone logs.

<Tabs>
<TabItem value="docker" label="Docker" default>

> Each container stores logs as a JSON file on the host machine. These can be shown through the

Reviewer: ```suggestion
Each container stores logs as a JSON file on the host machine. These can be shown with the
```

`docker logs {CONTAINER}` command. Log files are found at

> `var/lib/docker/containers/{CONTAINER_ID}/{CONTAINER_ID}-json.log`.

Reviewer: ```suggestion
`var/lib/docker/containers/{CONTAINER_ID}/{CONTAINER_ID}-json.log` by default.
```

psql -d firezone -h 127.0.0.1 -U postgres -c "UPDATE configurations SET saml_providers = '{}'"
```

> ## Debugging portal websocket connectivity issues

Reviewer: Proper noun

```suggestion
## Debugging portal WebSocket connectivity issues
```

51820/udp (v6)             ALLOW IN    Anywhere (v6)
```

## Locked Out of Admin Access

If you are locked out of the admin account for your Firezone instance

> you can re-create your admin user.

Reviewer: ```suggestion
you can recreate your admin user.
```

<Tabs>
<TabItem value="docker" label="Docker">

> Run the following command in the
> home directory of your Firezone installation. This is where you can
> find the `docker-compose.yml` and `.env` files.
> 
> ```shell
> docker compose exec firezone bin/create-or-reset-admin
> ```
> 
> This will reset the password on admin user using the default password
> found in the `.env` environment file.

Reviewer: ```suggestion
    First change directory to your Firezone root, then run the `bin/create-or-reset-admin` script
    to reset the admin user's password. The password for the user specified by 
  `ADMIN_EMAIL` 
    in `$HOME/.firezone/.env` will be reset to the `DEFAULT_ADMIN_PASSWORD` variable.
  
    ```shell
      cd $HOME/.firezone
      docker compose exec firezone bin/create-or-reset-admin
    ```

    **Note**: If local authentication is disabled, resetting the admin user's password will not 
    re-enable it.
```

</TabItem>

<TabItem value="omnibus" label="Omnibus">

Run the following command to reset the password on the default admin user.

```shell

> firezone-ctl create-or-reset-admin

Reviewer: ```suggestion
sudo firezone-ctl create-or-reset-admin
```

```

</TabItem>
</Tabs>

## Re-enable Local Authentication in CLI

For production deployments, we recommend adding a TOTP-based second factor to admin accounts.
If you promoted an account authenticated through an identity provider, you can consider
disabling local authentication for additional security.

> If errors occur with your identity provider integration, you can be locked out of
> access to the admin portal. You can un the following query on Firezone's Postgres database
> to re-enable local authentication.
> 
> ```SQL
> UPDATE configurations SET local_auth_enabled = 't'
> ```

Reviewer: ```suggestion
If issues arise with your identity provider integration, it's possible you could be locked out of the admin portal. To re-enable local authentication so you can log in and resolve the issue, run the following query on the host of your Firezone instance:

   ```shell
   cd $HOME/.firezone
   docker compose exec postgres psql -U postgres -h 127.0.0.1 -d firezone -c "UPDATE configurations SET local_auth_enabled = 't'"
   ```
```

51820/udp (v6)             ALLOW IN    Anywhere (v6)
```

> ## Locked out of admin access

Reviewer: ```suggestion
## Admin login isn't working
```

> If you are locked out of the admin account for your Firezone instance
> you can recreate your admin user.

Reviewer: ```suggestion
If the password for the account with email `ADMIN_EMAIL` isn't working, you can reset it using the process below.
```
